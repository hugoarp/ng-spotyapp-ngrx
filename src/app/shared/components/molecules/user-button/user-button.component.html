<section class="user-button">
  <img [src]="user.image" alt="Foto de perfil" class="user-button__img" />
  <p class="user-button__name">{{ user.name }}</p>
  <section class="user-button__dropdown">
    <article class="user-button__menu">
      <a class="user-button__menu-a" (click)="logout()">Cerrar sesiÃ³n</a>
    </article>
  </section>
</section>
