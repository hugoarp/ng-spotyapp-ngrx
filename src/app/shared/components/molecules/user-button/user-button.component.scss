@import "variables";

.user-button {
  background-color: $dark-secondary;
  border-radius: 1rem;
  display: flex;
  align-items: center;
}
