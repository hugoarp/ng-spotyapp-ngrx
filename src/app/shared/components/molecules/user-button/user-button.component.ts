import { Component, Input, OnInit } from '@angular/core';
import { Router } from '@angular/router';
import { UserInfo } from 'src/app/core/interfaces/user';

@Component({
  selector: 'app-user-button',
  templateUrl: './user-button.component.html',
  styleUrls: ['./user-button.component.scss'],
})
export class UserButtonComponent implements OnInit {
  @Input() user: UserInfo = {
    name: '',
    image: '',
  };
  constructor(private router: Router) {}

  ngOnInit(): void {}

  logout() {
    sessionStorage.removeItem('token');
    this.router.navigate(['/login']);
  }
}
