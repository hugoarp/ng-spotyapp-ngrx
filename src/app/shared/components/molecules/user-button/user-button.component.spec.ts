import { ComponentFixture, TestBed } from '@angular/core/testing';
import { UserButtonComponent } from './user-button.component';

describe('UserButtonComponent', () => {
  let component: UserButtonComponent;
  let fixture: ComponentFixture<UserButtonComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [UserButtonComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(UserButtonComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe hacer match con el snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });
});
