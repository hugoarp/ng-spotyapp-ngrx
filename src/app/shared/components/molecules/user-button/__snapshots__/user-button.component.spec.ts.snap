// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserButtonComponent debe hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <section
    class="user-button"
  >
    <img
      alt="Foto de perfil"
      class="user-button__img"
      src=""
    />
    <p
      class="user-button__name"
    >
      
    </p>
    <section
      class="user-button__dropdown"
    >
      <article
        class="user-button__menu"
      >
        <a
          class="user-button__menu-a"
        >
          Cerrar sesi√≥n
        </a>
      </article>
    </section>
  </section>
</div>
`;
